appendix/single-sign-on.rst

```diff
@@ -0,0 +1,521 @@
+
+2b. Install Apache
+^^^^^^^^^^^^^^^^^^
+
+.. code-block:: sh
+
+   # Ubuntu & Debian
+   $ apt update
+   $ apt install apache2 krb5-user libapache2-mod-auth-kerb
+
+   # CentOS
+   $ yum install httpd krb5-workstation mod_auth_kerb
+
+   # openSUSE
+   $ zypper ref
+   $ zypper install apache2 krb5-client apache2-mod_auth_kerb
+
+2c. Enable Apache modules
+^^^^^^^^^^^^^^^^^^^^^^^^^
+
+.. code-block:: sh
+
+   # Ubuntu, Debian, & openSUSE
+   $ a2enmod auth_kerb headers rewrite proxy proxy_html proxy_http proxy_wstunnel
+
+On systems without ``a2enmod`` (*e.g.,* CentOS),
+add/uncomment the appropriate ``LoadModule`` statements
+in your Apache config:
+
+.. code-block::
+
+   # /etc/httpd/conf/httpd.conf
+
+   LoadModule auth_kerb_module /usr/lib/apache2/modules/mod_auth_kerb.so
+   LoadModule headers_module modules/mod_headers.so
+   LoadModule rewrite_module modules/mod_rewrite.so
+   LoadModule proxy_module modules/mod_proxy.so
+   LoadModule proxy_html_module modules/mod_proxy_html.so
+   LoadModule proxy_http_module modules/mod_proxy_http.so
+   LoadModule proxy_wstunnel_module modules/mod_proxy_wstunnel.so
+
+2d. Configure Kerberos
+^^^^^^^^^^^^^^^^^^^^^^
+
+Kerberos realm configuration is how you tell the Zammad server
+about the *domain controller* (Active Directory server).
+
+.. note:: Replace the following placeholders in the sample config below:
+
+   :``<domain>``:                   Windows domain
+   :``<domain-controller>``:        Domain controller IP/FQDN(s)
+   :``<master-domain-controller>``: Master domain controller IP/FQDN
+
+                                    (must not be read-only,
+                                    but can be the same as ``<domain-controller>``)
+
+.. code-block::
+
+   # /etc/krb5.conf
+
+   [libdefaults]
+     default_realm = <domain>
+     default_tkt_enctypes = aes256-cts-hmac-sha1-96
+     default_tgs_enctypes = aes256-cts-hmac-sha1-96
+     permitted_enctypes = aes256-cts-hmac-sha1-96
+
+     kdc_timesync = 1
+     ccache_type = 4
+     forwardable = true
+     proxiable = true
+     fcc-mit-ticketflags = true
+
+   [realms]
+           # multiple KDCs ok (one per line)
+           <domain> = {
+                   kdc = <domain-controller>
+                   admin_server = <master-domain-controller>
+                   default_domain = <domain>
+           }
+
+   [domain_realm]
+           .<domain> = <domain>
+           <domain> = <domain>
+
+.. _sso-generate-keytab:
+
+2e. Generate keytab
+^^^^^^^^^^^^^^^^^^^
+
+Apache needs a Kerberos *keytab* (key table)
+to manage its shared secrets with the domain controller.
+
+
+.. note:: Replace the following placeholders in the commands below:
+
+   :``<zammad-host>``: Zammad FQDN
+   :``<domain>``:      Windows domain
+   :``<secret-key>``:  Secret key (**omit the leading** ``0x``)
+   :``<vno>``:         Secret key version number
+
+   The secret key and version number were found in :ref:`sso-register-spn` above.
+
+.. code-block:: sh
```

Comment:
Using `sh` syntax highlighting here highlights the word `for` in the third line (which is bad). The benefit is that it also highlights the code comments. Not sure how much you care either way; I'm on the fence and would be fine removing all syntax highlighting from this code sample.

Reply:
Would be fine to me, it's shell content.

Reply:
Okay great.

Technically, a lot of it is the `ktutil` interactive prompt, which is terminal/cli but not exactly the shell, but there's no sense in splitting hairs. :joy: